gst/tensor_decoder/tensordec-imagelabel.c

```diff
@@ -0,0 +1,313 @@
+      return TRUE;
+    else
+      return FALSE;
+      /** @todo Do not die for this */
+  }
+
+  GST_INFO ("Property mode-option-%d is ignored", opNum + 1);
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's TensorDecDef callback */
+static GstCaps *
+_getOutputDim (GstTensorDec * self, const GstTensorConfig * config)
+{
+  const uint32_t *dim;
+  int i;
+
+  g_return_val_if_fail (config != NULL, NULL);
+
+  dim = config->info.dimension;
+  /* This allows N:1:1:1 only! */
+  for (i = 1; i < NNS_TENSOR_RANK_LIMIT; i++)
+    if (dim[i] != 1) {
+      GST_ERROR ("Dimention %d is not 1, %u", i, dim[i]);
+      return NULL;
+    }
+
+  return gst_caps_from_string (GST_TENSOR_TEXT_CAPS_STR);
+}
+
+/** @brief tensordec-plugin's TensorDecDef callback */
+static gsize
+_getTransformSize (GstTensorDec * self, GstCaps * caps,
+    gsize size, GstCaps * othercaps, GstPadDirection direction)
+{
+  return 0;
+  /** @todo Use max_word_length if that's appropriate */
+}
+
+/** @brief Search for max. Macro for tensor_element union */
+#define search_max(type, i, max_index, max_val, bpe, data, num_data) \
+do {\
+  int i;\
+  type *cursor = (type *) data;\
+  max_val = cursor[0];\
+  max_index = 0;\
+  for (i = 1; i < num_data; i++) {\
+    if (cursor[i] > max_val) {\
+      max_val = cursor[i];\
+      max_index = i;\
+    }\
+  }\
+} while (0);
+
+/** @brief Shorter case statement for search_max */
+#define search_max_case(type, typename) \
+case typename:\
+  search_max(type, i, max_index, max_val._##type, bpe, input_data, num_data);\
+  break;
+
+
+/** @brief tensordec-plugin's TensorDecDef callback */
+static GstFlowReturn
+_decode (GstTensorDec * self, const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  ImageLabelData *data = self->plugin_data;
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+
+  gsize bpe = tensor_element_size[self->tensor_config.info.type];
+  tensor_element max_val;
+  guint max_index = 0;
+  gsize num_data;               /* Size / bpe */
+  void *input_data;
+
+  gsize size;
+  gchar *str;
+
+  g_assert (bpe > 0);
+  g_assert (outbuf);
+
+  input_data = input->data;
+  num_data = gst_tensor_info_get_size (&self->tensor_config.info) / bpe;
+
+  switch (self->tensor_config.info.type) {
+      search_max_case (int32_t, _NNS_INT32);
+      search_max_case (uint32_t, _NNS_UINT32);
+      search_max_case (int16_t, _NNS_INT16);
+      search_max_case (uint16_t, _NNS_UINT16);
+      search_max_case (int8_t, _NNS_INT8);
+      search_max_case (uint8_t, _NNS_UINT8);
+      search_max_case (double, _NNS_FLOAT64);
+      search_max_case (float, _NNS_FLOAT32);
+      search_max_case (int64_t, _NNS_INT64);
+      search_max_case (uint64_t, _NNS_UINT64);
+    default:
+      return GST_FLOW_NOT_SUPPORTED;
+  }
+
+  g_assert (max_index < data->total_labels);
+
+  /** @todo With option-2, allow to change output format */
+  str = g_strdup_printf ("%s", data->labels[max_index]);
+  GST_ERROR ("%s / %d", str, max_val._uint8_t);
```

Comment:
GST_ERROR -> GST_INFO or other log level.

Also please check max_val._uint8_t. (self->tensor_config.info.type, or remove this line ?)

Reply:
It was printing-debug instrumentation. I'll remove this.